README.md

```diff
@@ -69,29 +69,41 @@ Only the `source` import source phase is specified by this proposal.
 
 ### Defining Module Source
 
-Module source is defined through the [ECMA-262 ES modules HostLoadImportedModule refactoring][],
-which moves the construction of module records to the host, and on which we can define a
-`[[ModuleSourceObject]]` custom object to be returned by module source phase that is compatible
-with module expressions.
+The object provided by the module source phase must be a subclass of the
+`AbstractModuleSource` object, defined by this specification to be a minimal
+object representation for a compiled modular resource.
 
 ### JS Module Source
 
-The type of the module source object for a JavaScript module is would depend
-on the [module source][] specification.
+For JavaScript modules, the module source phase is then specified to return
+a `ModuleSource` object:
 
-In the current specification, the JS module source case just throws an error
-to support this as a future addition.
+```
+class ModuleSource extends AbstractModuleSource {
+  constructor () {
+    throw new Error();
+  }
+  get [Symbol.toStringTag] () {
+    if (this instanceof ModuleSource)
```

Comment:
the semantics should match a private field, and testing whether that field is `in this`, rather than `instanceof`.

also, shouldn't this be defined in AbstractModuleSource, and the ModuleSource implementation more look like:
```js
#kind = 'ModuleSource'
```
thus deferring everything to the AbstractModuleSource getter?

Reply:
I've changed the check in `Symbol.toStringTag` for `ModuleSource` to explicitly verify the host internal slot of `[[HostSourceData]]`.